src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -319,6 +320,63 @@ mgp_value_type FromTypedValueType(memgraph::query::TypedValue::Type type) {
 }
 }  // namespace
 
+bool MgpGraphIsTransactional(const mgp_graph &graph) noexcept {
+  auto storageMode = memgraph::storage::StorageMode::IN_MEMORY_TRANSACTIONAL;
+  std::visit(memgraph::utils::Overloaded{
+                 [&storageMode](memgraph::query::DbAccessor *dbAcc) { storageMode = dbAcc->GetStorageMode(); },
+                 [&storageMode](memgraph::query::SubgraphDbAccessor *dbAcc) { storageMode = dbAcc->GetStorageMode(); }},
+             graph.impl);
+  return storageMode != memgraph::storage::StorageMode::IN_MEMORY_ANALYTICAL;
+}
+
+bool IsDeleted(mgp_vertex *vertex) { return vertex->getImpl().impl_.vertex_->deleted; }
+
+bool IsDeleted(mgp_edge *edge) { return edge->impl.impl_.edge_.ptr->deleted; }
+
+bool ContainsDeleted(mgp_path *path) {
+  for (auto &vertex : path->vertices) {
+    if (IsDeleted(&vertex)) return true;
+  }
+  for (auto &edge : path->edges) {
+    if (IsDeleted(&edge)) return true;
+  }
+  return false;
+}
+
+bool ContainsDeleted(mgp_list *list) {
+  for (auto &elem : list->elems) {
+    if (ContainsDeleted(&elem)) return true;
+  }
+  return false;
+}
+
+bool ContainsDeleted(mgp_map *map) {
+  for (auto &[_, map_value] : map->items) {
+    if (ContainsDeleted(&map_value)) return true;
+  }
+  return false;
+}
+
+bool ContainsDeleted(mgp_value *val) {
+  switch (val->type) {
+    case MGP_VALUE_TYPE_VERTEX:
+      return IsDeleted(val->vertex_v);
+    case MGP_VALUE_TYPE_EDGE:
+      return IsDeleted(val->edge_v);
+    case MGP_VALUE_TYPE_PATH:
+      return ContainsDeleted(val->path_v);
+    case MGP_VALUE_TYPE_LIST:
+      return ContainsDeleted(val->list_v);
+    case MGP_VALUE_TYPE_MAP:
+      return ContainsDeleted(val->map_v);
+    // other types are primitives
```

Comment:
In case we add a new type which is not primitive, we can add here all types and exception will be thrown on compiler that case is not covered

Reply:
This is good from the maintainability perspective, adding it